jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/ResourceCacheTest.java

```diff
@@ -112,67 +114,75 @@ public void testMultipleSources1() throws Exception
+            MimeTypes mime = new MimeTypes();
 
-        CachedContentFactory rc3 = new CachedContentFactory(null, r.get(2), mime, false, false, CompressedContentFormat.NONE);
-        CachedContentFactory rc2 = new CachedContentFactory(rc3, r.get(1), mime, false, false, CompressedContentFormat.NONE);
-        CachedContentFactory rc1 = new CachedContentFactory(rc2, r.get(0), mime, false, false, CompressedContentFormat.NONE);
+            CachedContentFactory rc3 = new CachedContentFactory(null, r.get(2), mime, false, false, CompressedContentFormat.NONE);
+            CachedContentFactory rc2 = new CachedContentFactory(rc3, r.get(1), mime, false, false, CompressedContentFormat.NONE);
+            CachedContentFactory rc1 = new CachedContentFactory(rc2, r.get(0), mime, false, false, CompressedContentFormat.NONE);
 
-        assertEquals(getContent(rc1, "1.txt"), "1 - one");
-        assertEquals(getContent(rc1, "2.txt"), "2 - two");
-        assertEquals(getContent(rc1, "3.txt"), "3 - three");
+            assertEquals(getContent(rc1, "1.txt"), "1 - one");
+            assertEquals(getContent(rc1, "2.txt"), "2 - two");
+            assertEquals(getContent(rc1, "3.txt"), "3 - three");
 
-        assertEquals(getContent(rc2, "1.txt"), "1 - two");
-        assertEquals(getContent(rc2, "2.txt"), "2 - two");
-        assertEquals(getContent(rc2, "3.txt"), "3 - three");
+            assertEquals(getContent(rc2, "1.txt"), "1 - two");
+            assertEquals(getContent(rc2, "2.txt"), "2 - two");
+            assertEquals(getContent(rc2, "3.txt"), "3 - three");
 
-        assertEquals(null, getContent(rc3, "1.txt"));
-        assertEquals(getContent(rc3, "2.txt"), "2 - three");
-        assertEquals(getContent(rc3, "3.txt"), "3 - three");
+            assertNull(getContent(rc3, "1.txt"));
+            assertEquals(getContent(rc3, "2.txt"), "2 - three");
+            assertEquals(getContent(rc3, "3.txt"), "3 - three");
+        }
     }
 
     @Test
     public void testUncacheable() throws Exception
     {
         Path basePath = createUtilTestResources(workDir.getEmptyPathDir());
 
-        ResourceCollection rc = new ResourceCollection(
-            Resource.newResource(basePath.resolve("one")),
-            Resource.newResource(basePath.resolve("two")),
-            Resource.newResource(basePath.resolve("three")));
-
-        List<Resource> r = rc.getResources();
-        MimeTypes mime = new MimeTypes();
+        List<Resource> resourceList = Stream.of("one", "two", "three")
+            .map(basePath::resolve)
+            .map(Resource::newResource)
+            .toList();
 
-        CachedContentFactory rc3 = new CachedContentFactory(null, r.get(2), mime, false, false, CompressedContentFormat.NONE);
-        CachedContentFactory rc2 = new CachedContentFactory(rc3, r.get(1), mime, false, false, CompressedContentFormat.NONE)
+        try (Resource.Mount rcMount = Resource.mountCollection(resourceList))
```

Comment:
Since the `Resource`s could be created, there's no need for mounting.

Reply:
Greatly simplified